glossary.md

```diff
@@ -0,0 +1,37 @@
+A Glossary of Stateless Ethereum
+---------------------------------
+*This glossary is a work in progress; contributions are welcome*
+
+**Beam Sync**. An alternative sync protocol developed by the Trinity team with similarities to what is imagined for Stateless Ethereum. The scheme works by executing a block first, and then requesting missing state data on demand in the form of a witness.
```

Comment:
I'm good removing the Trinity reference as well.

Removing the `in the form of a witness` from this sentence seems like it may reduce confusion since beam-sync doesn't actually build witnesses or operate on them in any form.